mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -322,47 +325,29 @@ public Flags getApplicableFlag(Mailbox mailbox) throws MailboxException {
                 imapUidDAO.insert(composedMessageIdWithMetaData));
     }
 
-    private Stream<UpdatedFlags> updateFlagsOnMessage(Mailbox mailbox, FlagsUpdateCalculator flagUpdateCalculator, MailboxMessage message) {
-        return tryMessageFlagsUpdate(flagUpdateCalculator, mailbox, message)
+    private Stream<UpdatedFlags> updateFlagsOnMessage(Mailbox mailbox, FlagsUpdateCalculator flagUpdateCalculator, MailboxMessage message, long newModseq) {
+        return tryMessageFlagsUpdate(flagUpdateCalculator, message, newModseq)
             .map(Stream::of)
-            .orElse(handleRetries(mailbox, flagUpdateCalculator, message));
-    }
-
-    private Optional<UpdatedFlags> tryMessageFlagsUpdate(FlagsUpdateCalculator flagUpdateCalculator, Mailbox mailbox, MailboxMessage message) {
-        try {
-            long oldModSeq = message.getModSeq();
-            Flags oldFlags = message.createFlags();
-            Flags newFlags = flagUpdateCalculator.buildNewFlags(oldFlags);
-
-            boolean involveFlagsChanges = !identicalFlags(oldFlags, newFlags);
-            long newModSeq = generateNewModSeqIfNeeded(mailbox, oldModSeq, involveFlagsChanges);
-
-            message.setFlags(newFlags);
-            message.setModSeq(newModSeq);
-            if (updateFlags(message, oldModSeq)) {
-                return Optional.of(UpdatedFlags.builder()
-                    .uid(message.getUid())
-                    .modSeq(newModSeq)
-                    .oldFlags(oldFlags)
-                    .newFlags(newFlags)
-                    .build());
-            } else {
-                return Optional.empty();
-            }
-        } catch (MailboxException e) {
-            throw Throwables.propagate(e);
-        }
-    }
-
-    private long generateNewModSeqIfNeeded(Mailbox mailbox, long oldModSeq, boolean involveFlagsChanges) throws MailboxException {
-        if (involveFlagsChanges) {
-            return modSeqProvider.nextModSeq(mailboxSession, mailbox);
+            .orElse(handleRetries(mailbox, flagUpdateCalculator, message, newModseq));
+    }
+
+    private Optional<UpdatedFlags> tryMessageFlagsUpdate(FlagsUpdateCalculator flagUpdateCalculator, MailboxMessage message, long newModSeq) {
+        long oldModSeq = message.getModSeq();
+        Flags oldFlags = message.createFlags();
+        Flags newFlags = flagUpdateCalculator.buildNewFlags(oldFlags);
+
+        message.setFlags(newFlags);
+        message.setModSeq(newModSeq);
+        if (updateFlags(message, oldModSeq)) {
```

Comment:
When and how do you check if the flag delta really result in a flag change. Because if there is no concrete flag change the spec specify modSeq must not move !